### PR TITLE
Create preperer when needed for reused pages across methods

### PR DESCRIPTION
**Pull request by vladbarosan**

### CODE REVIEW

File: src/Model/PageTypeGo.cs [hunk 1]
```diff
@@ -133,5 +133,18 @@ public override int GetHashCode()
         {
             return Name.GetHashCode();
         }
+
+        /// <summary>
+        /// Updates the page with properties from the passed page. Throws an exception of pages are not equal.
```
Comment (jhendrixMSFT) [inline src/Model/PageTypeGo.cs]: Nit: The `of` should be `if`.

File: src/Model/PageTypeGo.cs [hunk 2]
```diff
@@ -133,5 +133,18 @@ public override int GetHashCode()
...
+
+        /// <summary>
+        /// Updates the page with properties from the passed page. Throws an exception of pages are not equal.
+        /// </summary>
+        /// <param name="page"></param>
+        public void MergeWithPage(PageTypeGo page)
+        {
+            if (!this.Equals(page))
+            {
+                throw new InvalidOperationException("CreatePageableTypeForMethod requires method to be a pageable operation");
```
Comment (jhendrixMSFT) [inline src/Model/PageTypeGo.cs]: I think the `CreatePageableTypeForMethod` should be `MergeWithPage` yes?

Comment (vladbarosan) [reply src/Model/PageTypeGo.cs]: ah yea will just use `nameof()`

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.118)

**Pull request merged**